web/src/engine/websites/CopyManga.ts

```diff
@@ -0,0 +1,161 @@
+import { FetchJSON } from '../FetchProvider';
+
+type APIResponse<T> = {
+    code: number,
+    message: string,
+    results: T
+}
+
+type APISingleComic = {
+    comic: APIComic
+}
+
+type APIResultList<T> = {
+    list: T[]
+}
+
+type APIPages = {
+    chapter: APIChapter
+}
+
+type APIComic = {
+    uuid: string,
+    path_word: string,
+    name: string
+}
+
+type APIChapter = {
+    uuid: string,
+    name: string,
+    contents: {
+        url: string
+    }[];
+    words: number[]
+}
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly API_HEADERS = {
+        'User-Agent': '"User-Agent" to "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.124 Safari/537.36 Edg/102.0.1245.44"',
+        version: new Date().toISOString().replace(/-/g, ".").split('T').shift(),
+        platform: "1",
+    };
+
+    private readonly apiurl = 'https://api.copymanga.org';
+
+    public constructor() {
+        super('copymanga', 'CopyManga', 'https://www.copymanga.site', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Media.Manga, Tags.Language.Chinese, Tags.Source.Aggregator);
+
+        //URL
+        this.Settings.url = new Text('urloverride', W.Plugin_Settings_UrlOverride, W.Plugin_Settings_UrlOverrideInfo, 'https://www.copymanga.site');
+        this.Settings.url.ValueChanged.Subscribe((_, value: string) => this.URI.href = value);
+        this.URI.href = this.Settings.url.value as string;
+
+        //image format
+        this.Settings.format = new Choice('image.format',
+            W.Plugin_Settings_ImageFormat,
+            W.Plugin_Settings_ImageFormatInfo,
+            'jpeg',
+            { key: 'jpeg', label: E.Settings_Global_DescramblingFormat_JPEG },
+            { key: 'webp', label: E.Settings_Global_DescramblingFormat_WEBP },
+        );
+
+        //"Use Global CDN"
+        this.Settings.useGlobalCDN = new Check(
+            'check-useGlobalCDN',
+            W.Plugin_CopyManga_Settings_GlobalCDN,
+            W.Plugin_CopyManga_Settings_GlobalCDNInfo,
+            true //true = 0, false = 1
+        );
+
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/comic/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const id = new URL(url).pathname.split('/').pop();
+        const request = this.createApiRequest(`api/v3/comic2/${id}`);
+        const data = await FetchJSON<APIResponse<APISingleComic>>(request);
+        return new Manga(this, provider, data.results.comic.path_word, data.results.comic.name.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList = [];
+        for (let page = 0, run = true; run; page++) {//its offset so page start at 0
```

Comment:
What is meant by offset in comment?
If its important, try to give a better explanation, otherwise remove the comment.

Reply:
Offset is when you use an item count instead of a page count.
Well its not really important